test/e2e/upgrade_test.go

```diff
@@ -208,26 +208,36 @@ func TestUpgrade(t *testing.T) {
 	tc, err := NewTestContext()
 	require.NoError(t, err)
 	err = tc.waitForConfiguredWindowsNodes(int32(numberOfMachineNodes), false, false)
```

Comment:
Please remind me why we did not check for the version annotation here i.e. before this PR.